### ISSUE TITLE
"Random Character," screen Unusable on Narrow Terminals

### ISSUE THREAD

**Issue by BlindGuyNW** [CleverRaven/Cataclysm-DDA#53699]
### Describe the bug

The "random character," screen narrows badly on terminals of 100 columns or so, though I'm not sure of the exact threshold. This makes it very difficult to play on screens which have a narrow width but are appropriately tall.

### Steps to reproduce

1. Open a terminal window with a width of around 100 columns.
2. GO through world generation and choose to create a random character.
3. Observe the resulting character display screen has important information cut off by the right edge of the window.

### Expected behavior

I expected the game to inform me if the terminal was too small, or to try to deal with the smaller window size some other way. I didn't expect information to be cut off without warning, especially since none of the other screens up to that point were missing anything significant.

### Versions and configuration

- OS: MacOs
    - OS Version: Darwin Zacharys-MacBook-Air.local 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64 x86_64
- Game Version: 0.F-4487-g9da956adf3 [64-bit]
- Graphics Version: Curses
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

This bug impacts blind users in particular, who often play on a Linux console which is hard to resize.

**Comment by BlindGuyNW** [comment]
It should also be clarified that I only used Mac OS for the initial report because it has an easily-configured window size. The problem applies across all platforms which have an Ncurses UI.
